src/main/java/fi/csc/chipster/jobhistory/JobHistoryResource.java

```diff
@@ -24,86 +25,123 @@
 import fi.csc.chipster.rest.hibernate.HibernateUtil;
 import fi.csc.chipster.rest.hibernate.Transaction;
 
-
 @Path("jobhistory")
 public class JobHistoryResource {
-	
+
 	private Config config;
 	private String jobDetail;
 	private HibernateUtil hibernate;
-	public static final String QUERY_PARAM_FILTER_ATTRIBUTE = "filter_attribute";
-	public static final String QUERY_PARAM_FILTER_VALUE = "filter_value";
-	
-	public JobHistoryResource(HibernateUtil hibernate,Config config){
-		this.config=config;
-		this.hibernate=hibernate;
+	public static final String FILTER_ATTRIBUTE_TIME = "Time";
+
+	public JobHistoryResource(HibernateUtil hibernate, Config config) {
+		this.config = config;
+		this.hibernate = hibernate;
 	}
-	
+
+	// http://localhost:8200/jobhistory?startTime=gt=2018-02-21T14:16:18.585Z
 	@GET
 	@Produces(MediaType.APPLICATION_JSON)
 	@Transaction
-	public Response getJobHistory(@Context UriInfo uriInfo){
-		
-		 MultivaluedMap<String, String> queryParams = uriInfo.getQueryParameters(); 
-		 Map<String,String> parameters = new HashMap<String,String>();
-
-		   for(String str : queryParams.keySet()){
-		     parameters.put(str, queryParams.getFirst(str));
-		   }
-	    System.out.println("parameters"+parameters);
-	    
-	    CriteriaBuilder builder=getHibernate().session().getCriteriaBuilder();
-		//Create CriteriaQuery
-		CriteriaQuery<JobHistoryModel> criteria=builder.createQuery(JobHistoryModel.class);
-		//Specify criteria root
-		Root<JobHistoryModel> root=criteria.from(JobHistoryModel.class);
-	
-		
-	
-	    if(parameters.size()>0){
-	    	//handle multiple filter attribute
-	    	//Constructing the list of parameters
-			List<Predicate> predicate=new ArrayList<Predicate>();
-			
-			for(String key: queryParams.keySet()){
-				System.out.println(key);
-				if(key!=null){
-					try{
-						System.out.println(parameters.get(key));
-						predicate.add(builder.equal(root.get(key),parameters.get(key)));
-					}catch(IllegalArgumentException e){
-						
+	public Response getJobHistory(@Context UriInfo uriInfo) {
+
+		MultivaluedMap<String, String> queryParams = uriInfo
+				.getQueryParameters();
+		Map<String, String> parameters = new HashMap<String, String>();
+
+		for (String str : queryParams.keySet()) {
+			parameters.put(str, queryParams.getFirst(str));
+		}
+		System.out.println("parameters" + parameters);
+
+		CriteriaBuilder builder = getHibernate().session().getCriteriaBuilder();
+		// Create CriteriaQuery
+		CriteriaQuery<JobHistoryModel> criteria = builder
+				.createQuery(JobHistoryModel.class);
+		// Specify criteria root
+		Root<JobHistoryModel> root = criteria.from(JobHistoryModel.class);
+
+		if (parameters.size() > 0) {
+			// handle multiple filter attribute
+			// Constructing the list of parameters
+			List<Predicate> predicate = new ArrayList<Predicate>();
+
+			for (String key : queryParams.keySet()) {
+				if (key != null) {
+					// Need to address gt and lt here
+					if (parameters.get(key).contains("gt")) {
```

Comment:
Consider refactoring to separate methods, because this is difficult to read with this many indentations. Or maybe it's just this github's view.

Reply:
the method at the moment is a bit messy, but refactoring in a different method will require many paramter passing. So, I am just leaving it like this and thinking about a better.